README.rst

```diff
@@ -20,33 +20,85 @@ configurations remain synchronized with the state of all databases in the
 cluster. PgQ is used for buffering mutation records until they are collected by
 the relay and forwarded to their final destination.
 
+Usage
+=====
+
+PGShovel is a tool which replicates mutations (``INSERT``, ``UPDATE`` and ``DELETE`` operations) made to a PostgreSQL data out of the database and into an ordered stream of transaction-aware mutation messages.  These messages can then be consumed by 3rd party systems as needed.  Commonly, the mutation stream is consumed by data warehouse operations, cache updating services, search index services or can even be processed into other new streams.
+
+At Disqus PGShovel is actively being developed to replicate changes from our master Postgres database - which contains the core commenting and other relational data - into our data warehousing system. There is further planned work to use the mutation stream to power a denormalized objects database, offering highly available low-latency access to object data.
+
 Concepts
 ========
 
+PGShovel has a few core concepts which are important to understanding it:
+
 Cluster
 -------
 
-A cluster contains replication sets.
+A Cluster contains a collection of Replication Sets.
 
 Replication Set
 ---------------
 
-Replication sets contain a collection of tables that are replicated together.
+Replication Sets contain configuration defining a collection of tables that are replicated together.
+
+Batch Operations
+----------------
+
+The smallest unit of data in PGShovel.  A Batch Operation is a message communicating one of 4 possible actions:
+
+1. BeginOperation
+2. MutationOperation
+3. CommitOperation
+4. RollbackOperation
+
+Ordered sequences of these operations describe, *with transactional awareness*, the mutations made against the tables contained in a Replication Set.
 
 Relay
 -----
 
-The relay consumes batches of mutations for the replication set and forwards
-those mutation batches to their destination.
+Relays are responsible for extracting queued raw mutations made against PostgreSQL, translating those raw mutation into Batch Operations and then logging (relaying) the Batch Operations to a final destination.  **There is exactly one Relay for each Replication Set**.
 
-.. todo:: Document replication worker, loader, target.
+Mutation Log
+------------
 
-Usage
-=====
+The Mutation Log is where database mutation events are buffered before being consumed by
+the Relay. Internally, PGShovel uses the PgQ tool to queue these mutations inside the database.
+
+Replication
+-----------
+
+The Replication system is a collection of tools which handle replicating a stream of Batch Operations into another Target, i.e. Cassandra or S3.
+
+Within Replication there are 4 main concepts:
+
+Loader
+~~~~~~
+
+The Loader is responsible for providing an interface to dump the initial data from the tables contained in a Replication Set.
```

Comment:
In my mind, the Loader (as an instance) isn't responsible for *providing* the interface, per se. (In my head, there's a distinction between the Loader as a thing that performs the load operation, and the Loader class as the definition of how the load should occur.)